ONSdigital/census-services-stub
# CDTT 397.3 Access with Unique Code

##Motivation and Context

Motivation for this is to implement the requirement given in CDTT-457 (CDTT-397.3) which is as follows:
```
GIVEN I have entered my Unique Access Code in Respondent Home 
WHEN I Submit the Unique Access Code 
AND the authentication is unsuccessful 
THEN I am presented with a message (TBA) informing me to XX (message TBA) 
```


##What has changed
The IAC Stub is changed now so that it only accepts the following 2 UAC codes as being valid: "vvvvvvvvvvvv" and "wwwwwwwwwwww". Any other code is treated as invalid, which means that a 404 status code will be returned.    

##How to test?
You need to run the census-rh-ui service, go to the port that RH runs on and enter an invalid UAC code. You should then see the error message shown in the screenshot below

##Links
[https://collaborate2.ons.gov.uk/jira/browse/CDTT-397]

##Screenshots (if appropriate):

![screen shot 2019-01-03 at 16 57 09](https://user-images.githubusercontent.com/24273056/50836878-36d9cc00-1352-11e9-8dd5-e815aa9331d6.png)
